### PR TITLE
Upgrade dependencies

### LINKED ISSUE #2954: [BUG] - Some cardano-cli queries do not work without "TestEnableDevelopmentNetworkProtocols" set to true
**Internal**

**Area**
`cardano-cli`

**Summary**

Query:  `cardano-cli query utxo --tx-in TX-IN`

There is an `ShelleyEncoderUnsupportedQuery` error when trying to use `cardano-cli query utxo --tx-in TX-IN` without adding "TestEnableDevelopmentNetworkProtocols" flag set to true in config file.

Before adding `"TestEnableDevelopmentNetworkProtocols"`:

```sh
cardano-cli query utxo --whole-utxo --testnet-magic 3 | grep ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     0        1000000 lovelace
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     1        5264604 lovelace

cardano-cli query utxo --tx-in ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986#1 
--testnet-magic 3

cardano-cli: ShelleyEncoderUnsupportedQuery (SomeSecond (GetUTxOByTxIn (fromList [TxInCompact (TxId {_unTxId = 
SafeHash "ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986"}) 0]))) ShelleyNodeToClientVersion3
```

After adding `"TestEnableDevelopmentNetworkProtocols": true` to config file:

```sh
cardano-cli query utxo --tx-in ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986#1 
--testnet-magic 3
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     1        5264604 lovelace
```

**Network**: Shelley QA

```sh
cardano-cli --version
cardano-cli 1.28.0 - linux-x86_64 - ghc-8.10
git rev 48429531f0d3d71fadce9a5971bf56a6df396f2d

cardano-node --version
cardano-node 1.28.0 - linux-x86_64 - ghc-8.10
git rev 48429531f0d3d71fadce9a5971bf56a6df396f2d
```
**Steps to reproduce**

1. start node `1.28.0` for `shelley_qa`:
```
nix-build -A scripts.shelley_qa.node -o shelley_qa-node-local
./shelley_qa-node-local/bin/cardano-node-shelley_qa
```
2. set `CARDANO_NODE_SOCKET_PATH` for `cardano-cli` and issue queries from above example without and with `"TestEnableDevelopmentNetworkProtocols": true` in config file.

**Expected behavior**
`"TestEnableDevelopmentNetworkProtocols": true` should not be needed for queries to work properly.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by newhoggy**
Resolves #2954

### CODE REVIEW

**Review by Jimbo4350 — APPROVED** [review]
LGTM!

**Comment by Jimbo4350** [comment]
We need to deal with that `TestEnableDevelopmentNetworkProtocols` stuff. I'll work on this now.

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs [hunk 1]
```diff
@@ -182,16 +187,7 @@ mkSomeConsensusProtocolCardano NodeByronProtocolConfiguration {
           -- version that this node will declare that it understands, when it
           -- is in the Mary era. Since Mary is currently the last known
```
Comment (nfrisby) [inline cardano-node/src/Cardano/Node/Protocol/Cardano.hs:188]: The last sentence of this comment is now out of date. It should be:

> That is, it is the version of protocol /after/ Mary, i.e. Alonzo.

**Review by nfrisby — APPROVED** [review]
The change to `ProtVer` looks correct to me. In particular, it continues the pattern of the previous eras.

My only GitHub comment was that there is a code comment that needs to be updated.

Thanks! Approving.

File: cabal.project [hunk 2]
```diff
@@ -217,8 +217,8 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: e9cda57df7ea6969edbc3bfc4e117668277d09c8
-  --sha256: 1a91z6yvlhzlqrf7fy51fmmpm3ssk684h6pjgg022cdlsq56yif2
+  tag: f36b37168b3790c5e3a1974994b04de188d8b0ae
```
Comment (nfrisby) [inline cabal.project:220]: This revision needs to be https://github.com/input-output-hk/ouroboros-network/commit/f06018002f5fd3d68569e194085d35159bebaebb in order for the `npcTestEnableDevelopmentHardForkEras` to no longer be required by Consensus. (IE the result of PR input-output-hk/ouroboros-network#3295)

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:188]: Good catch :+1:

**Review by dcoutts — COMMENTED** [review]
LGTM

**Pull request merged**